### ISSUE TITLE
option to specify cache directory

### ISSUE THREAD

**Issue by borisovg** [istanbuljs/nyc#684]
I am running `nyc` from within my `ROOT/tests/` directory (which has a separate private `package.json` file that pulls in all test related libraries and I use `--cwd ../` option when running tests.

After tests are finished there is `ROOT/node_modules/.cache/nyc` directory created (i.e. outside of `ROOT/tests`) which is annoying for apps that don't depend on anything and also apps that commit dependency code to git for various annoying operational reasons.

Please make the cache directory location configurable, similar to `--temp-directory`.

**Comment by bcoe** [comment]
@borisovg I believe you already should be able to configure the cache directory using `--cache-dir`, although it's not advertised in the options output.

Assuming this works for you, I'd happily take a pull request that documents it.

**Comment by borisovg** [comment]
@bcoe I've just tried that option and now the cache directory is not being created at all, but tests run without issue.

Here's the command I that I am running:
```
./node_modules/.bin/nyc \
        --all \
        --cwd ../ \
        --exclude tests/** \
        --reporter=html \
        --reporter=text \
        --cache-dir ./tests/.nyc_cache \
        --temp-directory ./tests/.nyc_output \
        node_modules/.bin/mocha *.spec.js
```

**Comment by borisovg** [comment]
And really the cleanest solution would be to by default create those directories relative to the current working directory (as opposed to the `--cwd`) as is already done for the `coverage` directory.

In my case it would mean they would be created within somewhere within `./tests` and I could drop the options entirely.

**Comment by bastianwegge** [comment]
Or you could use the `.nycrc` file and include 

```
{
  "report-dir": ".tmp/istanbul",
  "temp-directory": ".tmp/.nyc_output"
}
```

for example.

**Comment by madarche** [comment]
I confirm that the `--cache-dir` cli option works. But the `.nycrc` doesn't support a corresponding `cache-dir` property.

**Comment by JaKXz** [comment]
Has this been addressed by #968?

**Comment by borisovg** [comment]
`--cache-dir`does the job
